integration_test.go

```diff
@@ -80,9 +82,64 @@ func argHandler(arg int32) (int32, error) {
 	}
 }
 
+func verifyBaggage(ctx context.Context) error {
+	span := opentracing.SpanFromContext(ctx)
+	if span == nil {
+		return errors.New("missing span")
+	}
+
+	spanCtx, ok := span.Context().(jaeger.SpanContext)
+	if !ok {
+		return errors.New("trace context is not from jaeger")
+	}
+	if !spanCtx.IsDebug() {
+		return errors.New("span context is not configured to submit traces")
+	}
+
+	val := span.BaggageItem("baggagekey")
+	if val == "" {
+		return errors.New("missing baggage")
+	} else if val != "baggagevalue" {
+		return errors.New("unexpected baggage")
+	}
+	return nil
+}
+
+func verifyThriftHeaders(ctx thrift.Context) error {
+	headers := ctx.Headers()
+	if val, ok := headers["headerkey"]; ok {
+		if val != "headervalue" {
+			return errors.New("unexpected header")
+		}
+	} else {
+		return errors.New("missing header")
+	}
+
+	return nil
+}
+
+func verifyYARPCHeaders(reqMeta yarpc.ReqMeta) error {
+	headers := reqMeta.Headers()
```

Comment:
nit: can we instead have a single `veriyHeaders(headers map[string]string) error` method, and pass in the `ctx.Headers()` or `reqMeta.Headers()`? That way we can avoid the duplication here